### ISSUE TITLE
Why not using built in tf batch norm?

### ISSUE THREAD

**Issue by squarenabla** [rwth-i6/returnn#70]
Why don't we just using an official tf layer https://www.tensorflow.org/api_docs/python/tf/layers/batch_normalization which has the same functionality. 
I need a NCHW support. In the official version, I just set axis=1 and I'm fine. In the pure returnn implementation it's impossible:
line  https://github.com/rwth-i6/returnn/blob/e7a529ae665027bea8d911c0955da2b431cd38f7/TFNetworkLayer.py#L823 throws an error, because it works only with NHWC. And it's a pain in a b**t and lots of time to fix this due to legacy, legacy, legacy...
Much easier to use the standard function.

**Comment by albertz** [comment]
We want to be more flexible.
We want to have our own implementation.
I'm not sure what you mean by legacy.
Using an own implementation is much easier for flexibility.
I'm not sure what the problem is in adding the needed support. This should be simple.

Anyway, this discussion does not really belong here in the issue tracker. Also, I explained you many of these things already. If you have questions, just write me.
